hypothesis/client
# Replace Help and Tutorial panels with single, preact panel

This PR replaces the previous implementations of tutorial and help panels with a single, dual-purpose help/tutorial panel, with updated, clearer UX and behavior.

## Previous Variant

Previously, there were two separate panels for what is now a single panel: “tutorial” and “help.” The tutorial panel showed up for all new users automatically:

<img width="470" alt="Screen Shot 2019-11-06 at 12 30 29 PM" src="https://user-images.githubusercontent.com/439947/68323076-aadc2200-0092-11ea-8f67-9921274ee23d.png">

Once it was closed, it never showed up again and could not be reopened in any other way. The instructions, typography and icons were inconsistent and out-of-date.

The help panel looked like this:

<img width="470" alt="Screen Shot 2019-11-06 at 12 30 56 PM" src="https://user-images.githubusercontent.com/439947/68323095-ba5b6b00-0092-11ea-95d9-76b3aeb85d18.png">

You could also have both the tutorial and help open at the same time, which was a bit funky and fugly:

<img width="466" alt="Screen Shot 2019-11-06 at 12 30 48 PM" src="https://user-images.githubusercontent.com/439947/68323108-c34c3c80-0092-11ea-8dd4-8517d5f7db32.png">

## New Variant

The new auto-displayed tutorial panel looks like this:

<img width="473" alt="Screen Shot 2019-11-06 at 12 32 24 PM" src="https://user-images.githubusercontent.com/439947/68323121-ca734a80-0092-11ea-87b8-32a5b79cacb5.png">

It has consistent styling and behavior with our other newer panels.

The help panel now initially launches into the “tutorial” view (see above). Clicking on “About this version ->” displays a second “sub-panel” which contains much of what was in the former help panel:

<img width="463" alt="Screen Shot 2019-11-06 at 12 33 19 PM" src="https://user-images.githubusercontent.com/439947/68323131-cfd09500-0092-11ea-89fe-2c9eb088fc84.png">

Interacting with the panel looks like this:

![new-tutorial-panel](https://user-images.githubusercontent.com/439947/68323190-f098ea80-0092-11ea-885e-af86d81b0717.gif)

Two links—“Help Topics” and “New Support Ticket” are always available. The “New Support Ticket” link goes to our support-ticket form and pre-fills system/version data:

<img width="882" alt="Screen Shot 2019-11-06 at 12 35 22 PM" src="https://user-images.githubusercontent.com/439947/68323135-d2cb8580-0092-11ea-9b88-13cc5e161bd0.png">

### To Test Auto-Opening of Tutorial

The tutorial panel will “auto-open” for users who have not yet dismissed it—thus setting a profile preference and preventing it from auto-displaying it on subsequent app launches.

You can hand-manipulate your local `h` database to allow the same user to see the panel again if it has been dismissed.

To do so:

1. Choose which local user you want to test as (any should work) and log in as that user to `h`/the client.
2. Open a terminal to your `h` directory and run `make sql`
3. Run this query: `UPDATE public.user SET sidebar_tutorial_dismissed = false WHERE username = '<your_username>';` (obviously replacing the username)
4. Reload the client for that user and you should see the panel appear

You can re-run that SQL query any time you want to see the panel again for that user after dismissing it.

Fixes https://github.com/hypothesis/product-backlog/issues/1031
Fixes https://github.com/hypothesis/client/issues/1203
